catalog/migrate/migrate.go

```diff
@@ -0,0 +1,333 @@
+		return nil
+	}
+	repositoryRe, err := regexp.Compile(expr)
+	if err != nil {
+		return err
+	}
+	m.repositoryRe = repositoryRe
+	return nil
+}
+
+func (m *Migrate) Run() error {
+	ctx := context.Background()
+	// logger := logging.FromContext(ctx)
+	fmt.Println("==> Listing repositories to import")
+	repos, err := m.listReposToImport(ctx)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("==> Total", len(repos), "repositories to import")
+	for i, repo := range repos {
+		fmt.Printf("==> Import (%d/%d): %s (storage namespace: %s, default branch: %s)\n", i+1, len(repos), repo.Name, repo.StorageNamespace, repo.DefaultBranch)
+		_, err := m.migrateRepository(ctx, repo)
+		if err != nil {
+			fmt.Println("ERROR:", err)
+			continue
+		}
+	}
+	return nil
+}
+
+func (m *Migrate) listReposToImport(ctx context.Context) ([]*catalog.Repository, error) {
+	var repos []*catalog.Repository
+	var after string
+	for {
+		reposPage, hasMore, err := m.mvccCataloger.ListRepositories(ctx, 100, after)
+		if err != nil {
+			return nil, err
+		}
+		for _, repo := range reposPage {
+			if m.repositoryRe == nil || m.repositoryRe.MatchString(repo.Name) {
+				repos = append(repos, repo)
+			}
+		}
+		if !hasMore || len(reposPage) == 0 {
+			break
+		}
+		after = reposPage[len(reposPage)-1].Name
+	}
+	return repos, nil
+}
+
+func (m *Migrate) migrateRepository(ctx context.Context, repository *catalog.Repository) (*graveler.RepositoryRecord, error) {
+	// get or create repository
+	repo, err := m.getOrCreateTargetRepository(ctx, repository)
+	if err != nil {
+		return nil, err
+	}
+
+	// scan mvcc repository commits by order and reproduce on entry catalog
+	err = m.migrateCommits(ctx, repo)
+	if err != nil {
+		return nil, fmt.Errorf("repository %s: %w", repo.RepositoryID, err)
+	}
+	return repo, nil
+}
+
+func (m *Migrate) getOrCreateTargetRepository(ctx context.Context, repository *catalog.Repository) (*graveler.RepositoryRecord, error) {
+	repoID := graveler.RepositoryID(repository.Name)
+	branchID := graveler.BranchID(repository.DefaultBranch)
+
+	// check and create repository if needed
+	repo, err := m.entryCatalog.GetRepository(ctx, repoID)
+	if errors.Is(err, graveler.ErrRepositoryNotFound) {
+		repo, err = m.entryCatalog.CreateRepository(ctx, repoID, graveler.StorageNamespace(repository.StorageNamespace), branchID)
+		if err != nil {
+			return nil, fmt.Errorf("create repository: %w", err)
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	// check default branch includes at least one commit - compatibility issue
+	branch, err := m.entryCatalog.GetBranch(ctx, repoID, branchID)
+	if err != nil {
+		return nil, err
+	}
+	m.lastCommit = branch.CommitID
+	if m.lastCommit == "" {
+		metaRangeID, err := m.entryCatalog.WriteMetaRange(ctx, repoID, newEmptyIterator())
+		if err != nil {
+			return nil, err
+		}
+		commitID, err := m.entryCatalog.CommitExistingMetaRange(ctx, repoID, branchID, *metaRangeID, initialCommitter, initialCommitMessage, nil)
```

Comment:
Let's have a more informative commit message.  Perhaps "Create empty new branch for import" or something.